File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        for (int num : nums) {
            st.insert(num);
        }

        int n = nums.size();
        int longestLength = 0;
        for (int i = 0; i < n; i++) {
            if (st.find(nums[i] - 1) == st.end()) {
                int current = nums[i];
                int length = 1;
                while (st.find(current + 1) != st.end()) {
                    current = current + 1;
                    length++;
                }
                longestLength = max(longestLength, length);
            }
        }
        return longestLength;
    }
};

File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        finish = len(nums) - 1
        can_reach = 0
        for i in range(len(nums)):
            if can_reach < i:
                return False
            
            can_reach = max(can_reach, i + nums[i])
            
            if can_reach >= finish:
                return True

        return True
            
